clang/include/clang/AST/Decl.h

```diff
@@ -2484,6 +2498,9 @@ class FunctionDecl : public DeclaratorDecl,
 
   void setDeletedAsWritten(bool D = true) { FunctionDeclBits.IsDeleted = D; }
 
+  /// Only valid if isDeletedAsWritten() returns true.
+  void setDeletedMessage(StringLiteral *Message);
+
```

Comment:
I really think this should just be an extra parameter on `setDeletedAsWritten` instead.

Reply:
Yes, I would prefer (my argument is that there is no scenario in which we would call setDeletedMessage without first calling setDeletedAsWritten, and we always have the information available.

Reply:
> there is no scenario in which we would call setDeletedMessage without first calling setDeletedAsWritten

That’s true; I was thinking we might want to set the message elsewhere at some point, but I think that I was remembering the situation before I included it it in the `DefaultedOrDeletedInfo`. I’ll move it to `setDeletedAsWritten()` and just give it a default argument.